python/rmm/_lib/memory_resource.pyx

```diff
@@ -135,6 +141,20 @@ cdef class CudaMemoryResource(DeviceMemoryResource):
         pass
 
 
+cdef class CudaAsyncMemoryResource(DeviceMemoryResource):
+    def __cinit__(self, device=None):
+        self.c_obj.reset(
+            new cuda_async_memory_resource()
```

Comment:
What should failure look like here? 

* Should we just let the C++ error propagate up?
* Do we want to wrap this call in a `try..except` and re-raise with more information?
* Do we want to call `driverGetVersion()` and duplicate the check for 11.2 in C++ and Python?

Reply:
What does the C++ error look like if someone tries to create this on CUDA 11.0?

Reply:
https://github.com/rapidsai/rmm/blob/branch-0.19/include/rmm/mr/device/cuda_async_memory_resource.hpp#L44-L54

Reply:
We may want to improve the message here: https://github.com/rapidsai/rmm/blob/branch-0.19/include/rmm/mr/device/cuda_async_memory_resource.hpp#L53 to say that it was compiled without support instead of just the generic error message

Reply:
I think this part of the macro deals specifically with CUDA version < 11.2 -- @harrism any thoughts here on a possibly more informative error message? This will directly be propagated up to Python users.

Reply:
"cudaMallocAsync not supported by the version of the CUDA Toolkit used for compilation"?  I don't want to say "... used to compile RMM" since RMM is header-only.

Reply:
Improved the error message based on your suggestion.

Reply:
You changed the wrong error message. :)